spring-cloud-task-batch/src/test/java/org/springframework/cloud/task/batch/listener/TaskBatchExecutionListenerTests.java

```diff
@@ -86,6 +86,36 @@ public void testAutobuiltDataSource() {
 		validateContext();
 	}
 
+	@Test(expected = AssertionError.class)
+	public void testNoAutoConfigurationEnabled() {
+		this.applicationContext = SpringApplication.run(new Class[] {JobConfiguration.class,
+				PropertyPlaceholderAutoConfiguration.class,
+				EmbeddedDataSourceConfiguration.class,
+				BatchAutoConfiguration.class,
+				TaskBatchAutoConfiguration.class}, new String[] {"--spring.cloud.task.batch.listener.enabled=false"});
+		validateContext();
+	}
+
+	@Test(expected = AssertionError.class)
+	public void testNoAutoConfigurationEnable() {
+		this.applicationContext = SpringApplication.run(new Class[] {JobConfiguration.class,
+				PropertyPlaceholderAutoConfiguration.class,
+				EmbeddedDataSourceConfiguration.class,
+				BatchAutoConfiguration.class,
+				TaskBatchAutoConfiguration.class}, new String[] {"--spring.cloud.task.batch.listener.enable=false"});
+		validateContext();
+	}
+
+	@Test(expected = AssertionError.class)
+	public void testNoAutoConfigurationBothDisabled() {
+		this.applicationContext = SpringApplication.run(new Class[] {JobConfiguration.class,
+				PropertyPlaceholderAutoConfiguration.class,
+				EmbeddedDataSourceConfiguration.class,
+				BatchAutoConfiguration.class,
+				TaskBatchAutoConfiguration.class}, new String[] {"--spring.cloud.task.batch.listener.enable=false --spring.cloud.task.batch.listener.enabled=false"});
+		validateContext();
+	}
+
```

Comment:
Can we get a test where the user explicitly sets the property value to true?

Reply:
Forgot to add comment here.  The PR was updated with that test.